src/libServer/LookupServer.cpp

```diff
@@ -1106,37 +1135,79 @@ Json::Value LookupServer::GetEthBlockCommon(const TxBlock& txBlock,
                                                  includeFullTransactions);
 }
 
-Json::Value LookupServer::GetTransactionReceipt(const std::string& txnhash) {
-  Json::Value ret;
+Json::Value LookupServer::GetEthTransactionReceipt(const std::string& txnhash) {
+  auto hash = txnhash;
+
+  if (hash[0] == '0' && hash[1] == 'x') {
```

Comment:
I don't think you have to strip out the prefix as it's done automatically by `TransactionHash`.

Reply:
Update: I see it's used at the later stage so it may be fine. However I'd recommend using Transaction/Block Hash types and do comparison on them instead of writing hand-written loops.

Reply:
I am just calling some other zil function `GetTransaction` and so satisfying its requirements (IIRC it breaks with the 0x prefix). Now I think of it, what we could/should do is to create a convenience function in DataProcessing that will strip off '0x' iff it exists, then we can do it cleanly when we need to. What do you think?

Reply:
Thanks for explanation. However, that's very interesting it breaks with `0x` prefix since `GetTransaction` internally creates immediately `TxnHash, which is of type dev::h256 == FixedHash<32>`. `FixedHash` itself, when instantiated with hex string uses:
`explicit FixedHash(std::string const& _s, ConstructFromStringType _t = FromHex, ConstructFromHashType _ht = FailIfDifferent): FixedHash(_t == FromHex ? fromHex(_s, WhenError::Throw) : dev::asBytes(_s), _ht) {}`
This calls underneath:
`bytes dev::fromHex(std::string const& _s, WhenError _throw)` which moves the index by 2 when hex strings has `0x` prefix. 
In general I think we should reduce any string comparisons and manual stripping of 0x and use built-in aliases for Block/Transaction/Other hashes. The only exception could be variable-length hex for contract code/calldata. What do you think?

Reply:
The code breaks not at this point but lower down when it checks the size of the string coming in (it throws). It is ok when you remove this checking code, so I have taken off the '0x' stripping and removed the checking code. Hashes that are too long or short fail in the same way as hashes that are not found, it seems, so all is well.

I would say that I am not a massive fan (nor very confident it works) of the TxnHash style, but it is good to be consistent :+1: